### PR TITLE
[ASEC-891] Escape idp error

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### 📋 Changes

Escape the `cause.error` and `cause.error_description` properties in addition to the `message` per security review.

### CODE REVIEW

File: src/utils/errors.ts [hunk 1]
```diff
@@ -1,23 +1,12 @@
...
-}
-
 /**
  * @ignore
  */
 export function appendCause(errorMessage: string, cause?: Error): string {
   if (!cause) return errorMessage;
   const separator = errorMessage.endsWith('.') ? '' : '.';
-  return `${errorMessage}${separator} CAUSE: ${htmlSafe(cause.message)}`;
+  return `${errorMessage}${separator} CAUSE: ${cause.message}`;
```
Comment (Widcket) [inline src/utils/errors.ts:9]: Shouldn't this keep escaping the cause message?
If not, we should amend the documentation in line 51.

File: src/utils/errors.ts [hunk 2]
```diff
@@ -1,23 +1,12 @@
 import { HttpError } from 'http-errors';
```
Comment (Widcket) [inline src/utils/errors.ts:2]: We should amend the documentation in line 51:

> **IMPORTANT** This underlying error is **not** escaped in any way

Comment (adamjmcgrath) [reply src/utils/errors.ts:2]: Good point [8e4ab04](https://github.com/auth0/nextjs-auth0/pull/864/commits/8e4ab04ca54b750c4cb1ca988855afec6cd9ffcf)

**Comment by adamjmcgrath** [comment]
Couple of things to check, will reopen

**Comment by adamjmcgrath** [comment]
6e7aa2a Also escaping `ApplicationError` (From `RPError`) and basically any error property that comes from a third party.

**Pull request merged**